### ISSUE TITLE
[Mailer] Can create many mailers like in SwiftMailer

### ISSUE THREAD

**Issue by michaljusiega** [symfony/symfony#32535]
**Description**  
Hello all,
the new Mailer component released in 4.3 is to replace the SwiftMailer component. Everything works as it should. I have a huge project that connects to many databases, many REST servers and many SMTP servers. Traditionally, according to the documentation, I use SwiftMailer to send e-mails and it looks like this:

```
swiftmailer:
    default_mailer: mailerone
    mailers:
        mailerone:
            transport: '%mailer_one_transport_protocol%'
            host:      '%mailer_one_transport_host%'
            username:  '%mailer_one_transport_user%'
            password:  '%mailer_one_transport_password%'
            spool: { type: memory }
            logging:   '%kernel.debug%'
        mailersecond:
            transport: '%mailer_second_transport_protocol%'
            host:      '%mailer_second_transport_host%'
            username:  '%mailer_second_transport_user%'
            password:  '%mailer_second_transport_password%'
            spool: { type: memory }
            logging:   '%kernel.debug%'
        mailerthird:
            transport: '%mailer_third_transport_protocol%'
            host:      '%mailer_third_transport_host%'
            username:  '%mailer_third_transport_user%'
            password:  '%mailer_third_transport_password%'
            spool: { type: memory }
            logging:   '%kernel.debug%'
        mailerfourth:
            transport: '%mailer_fourth_transport_protocol%'
            host:      '%mailer_fourth_transport_host%'
            username:  '%mailer_fourth_transport_user%'
            password:  '%mailer_fourth_transport_password%'
            spool: { type: memory }
            logging:   '%kernel.debug%'

```

Unfortunately, I can not transform the implementation into a new component because it lacks the declaration of many Transports...

```
framework:
    mailer:
        dsn: '%env(MAILER_DSN)%'
```

**Example**  
I do not know if this is a good idea, but I think many people also have or will have a similar problem.

```
framework:
    mailer:
        default_mailer: mailerone
        mailers:
            mailerone: '%env(MAILER_ONE_DSN)%'
            mailersecond: '%env(MAILER_SECOND_DSN)%'
```

Sorry for my english, is so bad. Thank you for reading my problem and at the same time suggesting adding functionality (actually, in my opinion huge).

**Comment by Koc** [comment]
Do you really need many mailer? Is round robin (`smtp://mailerone && smtp://mailersecond`) and failover (`smtp://mailerone || smtp://mailersecond`) syntax inside dns is not enough?

**Comment by bambucha4** [comment]
I dont try this syntax. Yes. I need many mailers. ;)

**Comment by xelan** [comment]
We have the same use case here - multiple SMTP servers are used for the communication with different systems. :+1: for this proposal.

**Comment by kiler129** [comment]
I don't want to say "me too", but this is very important functionality in our case too. We send different e-mails via different gateways depending on the information contained. One of the examples is sending encrypted e-mails via internal network (government requirements) vs. delivering non-secure e-mails overs public networks.

**Comment by bambucha4** [comment]
> 
> 
> Do you really need many mailer? Is round robin (`smtp://mailerone && smtp://mailersecond`) and failover (`smtp://mailerone || smtp://mailersecond`) syntax inside dns is not enough?

I have reviewed the implementation Transport::fromDsn and I noticed that the && special characters send messages through all SMTP servers and || sends the minimum to the first or the next if the previous one failed. I found a pseudo-solution, but a really professional approach would be useful;)

**Comment by fabpot** [comment]
Implemented in #33409

**Comment by michaljusiega** [comment]
@fabpot you are awesome, thank you very much !

**Comment by mdeboer** [comment]
> Implemented in #33409

Is there any particular reason why was chosen to use an argument over a named autowire alias (don't know the exact name), as was done with the messenger component for example. You won't break BC either and refactoring (later on) would be a lot easier as instead of changing all function calls you would change the argument name in the constructor/wither/setter.

Just wondering :)
